src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Payment.xml

```diff
@@ -53,6 +53,19 @@
                     <attribute name="summary">Completes Payment</attribute>
                 </attribute>
             </itemOperation>
+
+            <itemOperation name="shop_change_payment_method">
```

Comment:
IMO you should use the same endpoint as in checkout, there is no need for adding the another one